From: Matthias Clasen <mclasen@redhat.com>
Date: Sat, 1 Apr 2017 23:39:07 -0400
Subject: Fix desktop file generation

We need to dist the .desktop.in file and replace @libexecdir@ last.

Origin: upstream, 0.7, commit:e67d8688cdcb820a39f39028f06f3cd058ccd6ba
---
 data/Makefile.am.inc | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/data/Makefile.am.inc b/data/Makefile.am.inc
index d59fc4f..0f60dbd 100644
--- a/data/Makefile.am.inc
+++ b/data/Makefile.am.inc
@@ -16,21 +16,23 @@ systemduserunit_DATA = xdg-desktop-portal-gtk.service
 CLEANFILES += $(systemduserunit_DATA)
 
 %.desktop.in: data/%.desktop.in.in
-	$(AM_V_GEN) sed -e "s|@libexecdir[@]|$(libexecdir)|" $< > $@
+	$(AM_V_GEN) msgfmt --desktop -d po --template $< -o $@
 
 %.desktop: %.desktop.in
-	$(AM_V_GEN) msgfmt --desktop -d po --template $< -o $@
+	$(AM_V_GEN) $(SED) -e "s|\@libexecdir\@|$(libexecdir)|" $< > $@
 
 desktopdir = $(datadir)/applications
 desktop_in_in_files = data/xdg-desktop-portal-gtk.desktop.in.in
-dist_desktop_DATA = xdg-desktop-portal-gtk.desktop
+desktop_in_files = xdg-desktop-portal-gtk.desktop.in
+desktop_DATA = xdg-desktop-portal-gtk.desktop
 
-CLEANFILES += $(dist_desktop_DATA)
+CLEANFILES += $(desktop_DATA)
 
 EXTRA_DIST += \
 	$(dbus_service_in_files)                \
         $(systemduserunit_in_files)             \
         $(desktop_in_in_files)                  \
+        $(desktop_in_files)                     \
         data/org.gtk.Notifications.xml          \
         data/org.gnome.SessionManager.xml       \
         data/org.gnome.Shell.Screenshot.xml     \
